src/Surfnet/StepupSelfService/SelfServiceBundle/Service/RemoteVetting/Value/AttributeCollection.php

```diff
@@ -45,9 +45,17 @@ private function add(Attribute $attribute)
         $this->attributes[] = $attribute;
     }
 
-
     public function getIterator()
     {
         return new ArrayIterator($this->attributes);
     }
+
+    public function getAttributes()
+    {
+        $output = [];
+        foreach ($this->attributes as $attribute) {
+            $output[$attribute->getName()] = $attribute->jsonSerialize();
```

Comment:
Why is serialization needed in a collection?

Reply:
Every collection, that implements the `AttributeCollectionInterface` should be able to do so. This way every implementation is responsible for how their data would be represented when displayed in the encrypted file. 

> Why is serialization needed in a collection?
The data that ends up in the encrypted file needs to be serializable, mainly to make things readable and structured in a recognizable manner. The collection is easily held responsible for this.

> Couldn't this be better moved to the IdentityData class itself with a dedicated serialization method? That way the collection is only responsible for the collection? Or even better a dedicated serializer.

That would be a possibility. But I judge against that. By utilizing the composite pattern this goal can be achieved in a more maintenance-friendly manner. Also, creating a dedicated serializer for this would create a very messy class that would constantly type check a collection and perform slightly different serialization logic for each collection.

Reply:
I did find that the `jsonEncode` call on the attribute should be removed. This will result in double encoding.

Reply:
If the serialization would be only on the IdentityData object it would be more maintenance friendly because then the code would be in one place. Certainly when the serialization is only needed for that object. The serialization could also be removed from the AttributeListDto object because it won't be needed there. That way double encoding would also be prevented. Maybe we could decide in the end if we want to refactor that. Thanks for elaborating!
Could you if you keep it this way change the name of the interface to for example `AttributeCollectionSerializationInterface` and the method to something like `getSerializedAttributes`.
Feel free to use an alternative the main reason is to clarify it's usage.